ttnn/cpp/ttnn/operations/embedding/device/kernels/dataflow/embeddings.cpp

```diff
@@ -3,145 +3,68 @@
+    const std::uint32_t weight_buffer_src_addr = get_arg_val<uint32_t>(1);
     const std::uint32_t batch_offset = get_arg_val<uint32_t>(2);
     const std::uint32_t weights_offset = get_arg_val<uint32_t>(3);
     const std::uint32_t num_blocks = get_arg_val<uint32_t>(4);
 
     const std::uint32_t index_idx = get_arg_val<uint32_t>(5);
 
-#define in_is_dram get_compile_time_arg_val(0) == 1
-#define in_stick_size_is_power_of_two get_compile_time_arg_val(1) == 1
-    constexpr uint32_t input_page_size = get_compile_time_arg_val(2);
-#if (in_stick_size_is_power_of_two)
-    constexpr uint32_t log_base_2_of_input_page_size = get_compile_time_arg_val(3);
-    const InterleavedPow2AddrGen<in_is_dram> input = {
-        .bank_base_address = input_dram_buffer_src_addr,
-        .log_base_2_of_page_size = log_base_2_of_input_page_size  // TODO(AP): refactor
-    };
-#else
-    const InterleavedAddrGen<in_is_dram> input = {
-        .bank_base_address = input_dram_buffer_src_addr, .page_size = input_page_size};
-#endif
+    constexpr uint32_t cb_id_in0 = get_compile_time_arg_val(0);
+    constexpr uint32_t cb_id_in1 = get_compile_time_arg_val(1);
+    constexpr uint32_t cb_id_in2 = get_compile_time_arg_val(2);
 
-#define weights_is_dram get_compile_time_arg_val(4) == 1
-#define weight_stick_size_is_power_of_two get_compile_time_arg_val(5) == 1
-    constexpr uint32_t weight_stick_size = get_compile_time_arg_val(6);
-#if (weight_stick_size_is_power_of_two)
-    constexpr uint32_t log_base_2_of_weights_page_size = get_compile_time_arg_val(7);
-    const InterleavedPow2AddrGen<weights_is_dram> weights = {
-        .bank_base_address = weights_dram_buffer_src_addr,
-        .log_base_2_of_page_size = log_base_2_of_weights_page_size  // TODO(AP): refactor
-    };
-#else
-    const InterleavedAddrGen<weights_is_dram> weights = {
-        .bank_base_address = weights_dram_buffer_src_addr, .page_size = weight_stick_size};
-#endif
+    constexpr bool input_in_dram = get_compile_time_arg_val(3) == 1;
+    constexpr uint32_t input_page_size = get_compile_time_arg_val(4);
+    const auto input = get_interleaved_addr_gen<input_in_dram, input_page_size>(input_buffer_src_addr);
 
-    constexpr uint32_t rows_per_block = get_compile_time_arg_val(8);
-    constexpr uint32_t input_block_size_bytes = get_compile_time_arg_val(9);
+    constexpr bool weight_in_dram = get_compile_time_arg_val(5) == 1;
+    constexpr uint32_t weight_stick_size = get_compile_time_arg_val(6);
+    const auto weights = get_interleaved_addr_gen<weight_in_dram, weight_stick_size>(weight_buffer_src_addr);
 
-    constexpr uint32_t cb_id_in0 = 0;
-    constexpr uint32_t cb_id_in1 = 1;
-    constexpr uint32_t cb_id_in2 = 2;
+    constexpr uint32_t rows_per_block = get_compile_time_arg_val(7);
+    constexpr uint32_t input_block_size_bytes = get_compile_time_arg_val(8);
 
-    constexpr uint32_t tile_height = 32;
+    prepare_local_cache(cb_id_in2, weights, weight_stick_size, 6);
```

Comment:
Please use a descriptive constexpr or indicate what the parameter is, e.g.`/*pad_token_arg_idx=*/6`